### PR TITLE
Test select2 autocomplete performance

### PR DESCRIPTION
**Pull request by chrfritsch**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-3.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: tests/src/Nightwatch/Tests/AutoCompleteField.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      }
+
+      browser.fieldAutoFill(fieldName, requiredFields[fieldName]);
+    });
+    browser.performance.endMark();
+
+    browser.performance
+      .startMark("finding of existing tag")
+      // First selection is the slowest because caches are not warm.
+      .select2.selectValue("field_22", "bund", 2, 10000)
```
Comment (mtodor) [inline tests/src/Nightwatch/Tests/AutoCompleteField.js:56]: I would measure the 1st search and 2nd search. For example, there could be a problem with 1st search (cold cache) and it suddenly takes longer for it to finish. If we are measuring 3 searches, then performance drop for 1st search could be reduced with successive searches.

That's why I would make only 2 searches and I would mark (measure) each of them.

File: tests/src/Nightwatch/Tests/AutoCompleteField.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      )
+      .drupalLogin({ name: "test-admin", password: "test-admin" })
+      .useXpath();
+
+    browser
+      .drupalRelativeURL(`/node/add/${bundle}`)
+      .waitForElementVisible('//*[@id="edit-submit"]', 1000);
+
+    // Fill required fields for content bundle.
+    browser.performance.startMark("fill required fields");
```
Comment (mtodor) [inline tests/src/Nightwatch/Tests/AutoCompleteField.js]: I think it's not relevant for this test to measure the time needed to fill irrelevant fields.

File: tests/src/Nightwatch/Tests/AutoCompleteField.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      .startMark("finding of existing tag")
+      // First selection is the slowest because caches are not warm.
+      .select2.selectValue("field_22", "bund", 2, 10000)
+      .select2.selectValue("field_22", "bund", 4, 10000)
+      .select2.selectValue("field_22", "bund", 6, 10000)
+      .performance.endMark();
+
+    // Submit form.
+    browser
+      .click('//*[@id="edit-submit"]')
```
Comment (mtodor) [inline tests/src/Nightwatch/Tests/AutoCompleteField.js]: I don't have strong opinion about saving of article for this test, just a thought. I'm not sure if the saving of an article is relevant for this test. If we would have a field with auto-create, then measuring article saving would be interesting for us, but currently, there is no need for it. What do you think?

Comment (chrfritsch) [reply tests/src/Nightwatch/Tests/AutoCompleteField.js]: I agree. We don't need to test that here

**Review by mtodor — APPROVED** [review]

**Pull request merged**